### PR TITLE
CI: Update `base-ipu` docker image

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
- Resolves timeout in master due to very slow dependency resolution by conda. (from `>100min` to `<10min`)
- Most of the change comes from the `base-cuda` dockerfile.

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: dockers/base-ipu/Dockerfile [hunk 1]
```diff
@@ -17,14 +17,14 @@ FROM ubuntu:20.04
 LABEL maintainer="Lightning-AI <https://github.com/Lightning-AI>"
 
 ARG PYTHON_VERSION=3.9
-ARG CONDA_VERSION=4.9.2
+ARG PYTORCH_VERSION=1.9
```
Comment (akihironitta) [inline dockers/base-ipu/Dockerfile:20]: I'm confused. How was it possible to build the docker image with PyTorch 1.9 without `ARG PYTORCH_VERSION`? https://github.com/Lightning-AI/lightning/blob/b25275ccc27652b91d85d49b7bc220b37c921b54/.github/workflows/cicd-pytorch_dockers.yml#L169

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Comment by akihironitta** [comment]
I'll keep this PR as draft so that we don't merge it before verifying it in the IPU CI.

File: dockers/base-ipu/Dockerfile [hunk 2]
```diff
@@ -37,59 +37,49 @@ RUN apt-get update -qq && \
...
+    # install assistant.py's dependency
     pip install -q fire && \
-    python requirements/adjust-versions.py requirements/extra.txt && \
-    # Install remaining requirements
-    pip install -r requirements/extra.txt --no-cache-dir && \
-    pip install -r requirements/test.txt --no-cache-dir && \
+    # disable cache \
+    pip config set global.cache-dir false && \
+    # set particular PyTorch version
+    python ./requirements/pytorch/adjust-versions.py requirements/pytorch/base.txt ${PYTORCH_VERSION} && \
```
Comment (Borda) [inline dockers/base-ipu/Dockerfile:72]: Have you dropped PT version?

File: dockers/base-ipu/Dockerfile [hunk 3]
```diff
@@ -37,59 +37,49 @@ RUN apt-get update -qq && \
...
+        python${PYTHON_VERSION}-dev \
     && \
-# Install conda and python.
-# NOTE new Conda does not forward the exit status... https://github.com/conda/conda/issues/8385
-    curl -o ~/miniconda.sh https://repo.anaconda.com/miniconda/Miniconda3-py39_${CONDA_VERSION}-Linux-x86_64.sh && \
-    chmod +x ~/miniconda.sh && \
-    ~/miniconda.sh -b && \
-    rm ~/miniconda.sh && \
-# Cleaning
+    update-alternatives --install /usr/bin/python${PYTHON_VERSION%%.*} python${PYTHON_VERSION%%.*} /usr/bin/python${PYTHON_VERSION} 1 && \
```
Comment (Borda) [inline dockers/base-ipu/Dockerfile:49]: Not sure if all HPU drivers are fully compatible with any Python version

File: dockers/base-ipu/Dockerfile (hunk 2 above)
Comment (akihironitta) [reply dockers/base-ipu/Dockerfile:72]: No, I haven't, I think. Do I need to?

File: dockers/base-ipu/Dockerfile (hunk 3 above)
Comment (akihironitta) [reply dockers/base-ipu/Dockerfile:49]: IMO, having it as a variable itself is fine.

**Comment by akihironitta** [comment]
> I'll keep this PR as draft so that we don't merge it before verifying it in the IPU CI.

This PR is basically ready, but I'd like to check in advance if the docker image built from this dockerfile works in our CI...

**Comment by otaj** [comment]
Hi, @akihironitta, what do you need help with? I'm here :slightly_smiling_face:

**Comment by akihironitta** [comment]
For the note, the only task left here is to validate the change in the IPU server:
- [ ] Replace the agent runner with the new docker image in the IPU CI server.

**Comment by Borda** [comment]
@akihironitta blocked in title does not do anything :)
so what needs to be done, publish the newly built image?

**Comment by justusschock** [comment]
@akihironitta is this still relevant? Otherwise feel free to close it :)

**Comment by akihironitta** [comment]
Closing this PR since #14934 fixed the original issue which this PR was trying to resolve.

**Pull request closed without merging**